# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# Filename: package/.../dietlibc/patches/pkg_hotplug2.patch
# Copyright (C) 2007 The OpenSDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Description: Unlike struct dirent64, struct dirent has no d_type on dietlibc.

--- ./linux24_compat/hotplug2-coldplug-2.4.c.orig	2007-10-05 03:42:18.000000000 +0200
+++ ./linux24_compat/hotplug2-coldplug-2.4.c	2007-10-05 03:51:16.000000000 +0200
@@ -160,7 +160,7 @@
 	
 	DIR *procdir, *pcidir;
 	FILE *item;
-	struct dirent *cur_bus, *cur_device;
+	struct dirent64 *cur_bus, *cur_device;
 	
 	procdir = opendir("/proc/bus/pci");
 	if (procdir == NULL) {
@@ -168,7 +168,7 @@
 		return;
 	}
 	
-	while ((cur_bus = readdir(procdir)) != NULL) {
+	while ((cur_bus = readdir64(procdir)) != NULL) {
 		if (cur_bus->d_type != DT_DIR)
 			continue;
 		
@@ -179,7 +179,7 @@
 		snprintf(pci_dir_name, 255, "/proc/bus/pci/%s", cur_bus->d_name);
 		pcidir = opendir(pci_dir_name);
 		
-		while ((cur_device = readdir(pcidir)) != NULL) {
+		while ((cur_device = readdir64(pcidir)) != NULL) {
 			if (cur_device->d_type != DT_REG)
 				continue;
 			
@@ -257,7 +257,7 @@
 	
 	DIR *procdir, *usbdir;
 	FILE *item;
-	struct dirent *cur_bus, *cur_device;
+	struct dirent64 *cur_bus, *cur_device;
 	
 	procdir = opendir("/proc/bus/usb");
 	if (procdir == NULL) {
@@ -265,7 +265,7 @@
 		return;
 	}
 	
-	while ((cur_bus = readdir(procdir)) != NULL) {
+	while ((cur_bus = readdir64(procdir)) != NULL) {
 		if (cur_bus->d_type != DT_DIR)
 			continue;
 		
@@ -276,7 +276,7 @@
 		snprintf(usb_dir_name, 255, "/proc/bus/usb/%s", cur_bus->d_name);
 		usbdir = opendir(usb_dir_name);
 		
-		while ((cur_device = readdir(usbdir)) != NULL) {
+		while ((cur_device = readdir64(usbdir)) != NULL) {
 			if (cur_device->d_type != DT_REG)
 				continue;
 			
